R/zchunk_LB109.ag_an_ALL_R_C_Y.R

```diff
@@ -33,61 +33,209 @@ module_aglu_LB109.ag_an_ALL_R_C_Y_DISABLED <- function(command, ...) {
+        mutate(NetExpAdjFrac = if_else(GlobalOtherUses_Mt == 0, 0, OtherUses_Mt_adj / GlobalOtherUses_Mt),
+               # Allocate the changes in global net exports (total of negative other uses) among regions with positive other uses
+               NetExp_Mt = NetExp_Mt_adj - NetExpAdjFrac * GlobalNetExpAdj,
+               # Rebuild the mass balance table
+               Supply_Mt = Prod_Mt - NetExp_Mt,
+               OtherUses_Mt = Supply_Mt - Food_Mt - Biofuels_Mt - Feed_Mt) %>%
+        gather(flow, value, -GCAM_region_ID, -GCAM_commodity, -year) %>%
+        # Select only the flow variables for primary agricultural goods
+        filter(flow %in% ag_Flow_cols) %>%
+        # Re-order the flow variables so the columns are in the right order
+        mutate(flow = factor(flow, levels = ag_Flow_cols),
+               value = round(value, aglu.DIGITS_CALOUTPUT)) %>%
+        spread(flow, value) ->
+        L109.ag_ALL_Mt_R_C_Y
+      }
+
+    # Part 2: Animal commodities
+    # List of all flows for animal products
+    an_Flow_cols <- c( "Prod_Mt", "NetExp_Mt", "Supply_Mt", "Food_Mt", "OtherUses_Mt" )
+    # Name the flows in each table
+    L105.an_Prod_Mt_R_C_Y %>%
+      mutate(flow = "Prod_Mt") %>%
+      # Combine all flow tables
+      bind_rows(mutate(L106.an_NetExp_Mt_R_C_Y, flow = "NetExp_Mt")) %>%
+      bind_rows(mutate(L105.an_Food_Mt_R_C_Y, flow = "Food_Mt")) %>%
+      # Convert to wide format for easier calculations
+      spread(flow, value) %>%
+      # Calculate the domestic supply
+      mutate(Supply_Mt = Prod_Mt - NetExp_Mt,
+             # Calculate other uses
+             OtherUses_Mt = Supply_Mt - Food_Mt) -> L109.an_ALL_Mt_R_C_Y
+
+    # Adjust global and regional animal product mass balances to remove net negative other uses
+    # Assign negative other net uses to imports, and adjust global trade to maintain balances
+    # Changes in global net exports are apportioned among regions with positive other uses, according to shares
+
+    # First calculate the changes in global net exports
+    if(any(L109.an_ALL_Mt_R_C_Y$OtherUses_Mt < 0)) {
+      L109.an_ALL_Mt_R_C_Y %>%
+        # Subset negative and positive "other uses" separately
+        mutate(NegOtherUses_Mt = if_else(OtherUses_Mt < 0, OtherUses_Mt, 0),
+               # Positive will be the new adjusted "other uses", and replace negative with zero
+               OtherUses_Mt_adj = if_else(OtherUses_Mt >= 0, OtherUses_Mt, 0),
+               # Assign negative "other uses" to imports, and calculate the adjusted regional net exports
+               NetExp_Mt_adj = NetExp_Mt + NegOtherUses_Mt) %>%
+        group_by(GCAM_commodity, year) %>%
+        # Calculate the changes in global net exports = sum of negative other uses, global other uses =  sum of positive other uses
+        summarise_at(vars(GlobalNetExpAdj = NegOtherUses_Mt, GlobalOtherUses_Mt = OtherUses_Mt_adj), sum) -> L109.an_ALL_Mt_glbl_C_Y
```

Comment:
Here and elsewhere, would it be clearer to say 
```R
  summarise(GlobalNetExpAdj = sum(NegOtherUses_Mt), 
            GlobalOtherUses_Mt = sum(OtherUses_Mt_adj))
```

I don't have a strong preference but consider.